test/test_shared.cpp

```diff
@@ -89,6 +91,32 @@ using unit_test::TestContext;
 // check-testcase` (or one of its friends) from the command line.
 
 
+#ifdef _WIN32
+namespace {
+    // This does NOT work like on POSIX: The child will begin execution from the unit
+    // test entry point, not from where fork() took place.
+    DWORD winfork(std::string unit_test_name)
+    {
+        if (getenv("REALM_FORKED"))
+            return GetCurrentProcessId();
+
+        remove("winfork.bat");
+        std::ofstream myfile("winfork.bat");
```

Comment:
Did you think about a solution here? You could either make the file name unique or use `NONCONCURRENT_TEST` when using this function otherwise we can get sporadic test failures when running the test suite with `UNITTEST_THREADS>1` if two tests that use winfork happen to run at the same time.